src/parser/syntax.rs

```diff
@@ -0,0 +1,745 @@
+use crate::field::LurkField;
+use nom::{
+    branch::alt,
+    bytes::complete::{tag, take_till},
+    character::complete::{anychar, char, multispace0, multispace1, none_of},
+    combinator::{opt, peek, success, value},
+    error::context,
+    multi::{many0, separated_list1},
+    sequence::{delimited, preceded, terminated},
+    IResult,
+};
+
+use crate::{
+    num::Num,
+    parser::{
+        base,
+        error::{ParseError, ParseErrorKind},
+        position::Pos,
+        string, Span,
+    },
+    symbol,
+    symbol::Symbol,
+    syntax::Syntax,
+    uint::UInt,
+};
+
```

Comment:
A type alias may help:
``` 
type ParseResult<'a, F, T> = IResult<Span<'a>, T, ParseError<Span<'a>, F>>;
```

Reply:
Not sure, `IResult` is a `nom` idiom (https://docs.rs/nom/latest/nom/type.IResult.html) that is itself already a type alias:

```
pub type IResult<I, O, E = error::Error<I>> = Result<(I, O), Err<E>>;
```

So while this would definitely clean up the code, I worry think it might add more friction to people learning the code to further rename it. I'll try it out though and see if I like it

Reply:
okay, tried it out `ParseResult` is a clear improvement